From 0e9d6a23fdc8ae34fea79f39e4a1b75276ba4056 Mon Sep 17 00:00:00 2001
From: tloan <tloan@phytec.com>
Date: Wed, 25 Feb 2026 22:34:58 +0000
Subject: [PATCH 2/2] lvgl v9.1: fix weston frame deadlock

- weston sometimes drops frame callbacks (signifying that a frame was done drawing)
- lvgl was waiting indeffinately for frame callbacks. dropped callback causeed a total freeze of the app
- configures a safer 50ms timeout so that if a callback is dropped lvgl continues running

Signed-off-by: tloan <tloan@phytec.com>
---
 src/drivers/wayland/lv_wayland.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/drivers/wayland/lv_wayland.c b/src/drivers/wayland/lv_wayland.c
index 3d72811a7..e4584a073 100644
--- a/src/drivers/wayland/lv_wayland.c
+++ b/src/drivers/wayland/lv_wayland.c
@@ -2813,7 +2813,11 @@ bool lv_wayland_timer_handler(void)
             LV_LOG_TRACE("The window is hidden or minimized");
 
             /* Simply blocks until a frame done message arrives */
-            poll(&application.wayland_pfd, 1, -1);
+	    int ret = poll(&application.wayland_pfd, 1, 50);
+            if(ret == 0) {
+                /* Wayland dropped the frame callback. Break the deadlock. */
+                window->frame_done = true;
+            }
 
             /* Resume lvgl on the next cycle */
             return false;
-- 
2.34.1

